### ISSUE TITLE
Add support for Thunderbird CalDAV/CardDAV automatic setup

### ISSUE THREAD

**Issue by strarsis** [mailcow/mailcow-dockerized#4307]
## Summary
Connecting Thunderbird with a Google Mail account seems to offer more features than usual IMAP/SMTP and manual configuration of calendars-/contact-specific protocols.

Thunderbird asks for additional OAuth authentication - afterwards it detects calendars and address books and configures itself to make use of them.
It also shows "End-to-end" encryption. What protocols are these? Proprietary?

## Motivation
It would be nice to have open, standardized protocols to offer the same features as proprietary/walled-garden solutions and user experience.

## Additional context
Add a new Google account to recent Thunderbird.

**Comment by mkuron** [comment]
This is how it currently works: Calendars and address books require an extension (SOGo Connector or TbSync), which cannot automatically be installed. If you already have SOGo Connector installed, it will automatically configure contacts and calendars for Mailcow. If you have TbSync installed, it can be configured with just a few clicks.

I didn't know this until today, but recent Thunderbird versions support CardDAV. Looking at https://github.com/mozilla/releases-comm-central/blob/e6dbce863262e1c938a06c6c66d52d22bdcb24a7/mail/components/accountcreation/content/accountSetup.js I can see they have some kind of automatic setup mechanism. That is what you are seeing with GMail. Perhaps we can support that with Mailcow too?

> Thunderbird asks for additional OAuth authentication

Thunderbird has a small list of providers for which it supports it, see https://github.com/mozilla/releases-comm-central/blob/master/mailnews/base/src/OAuth2Providers.jsm. More information at https://dovecot.org/list/dovecot/2021-June/122400.html. Dovecot has added OAuth2 support recently, but it's obviously not supported with Thunderbird.

> It also shows "End-to-end" encryption.

That just takes you to the S/MIME and PGP screen in the account settings.

So in summary, the only thing we can try to do is support Thunderbird's new automatic DAV setup.

**Comment by mkuron** [comment]
Turns out that already works and looks exactly like with Gmail, sans the OAuth part. I will update the client documentation accordingly.
<img width="859" alt="Screen Shot 2021-10-31 at 16 54 13" src="https://user-images.githubusercontent.com/1748330/139592173-da755ab1-552b-4919-ba4e-937b75b4678a.png">

**Comment by pReya** [comment]
@mkuron These calendars/address books are not showing up for me. What am I doing wrong? Where/how did you start the wizard to show these individual items with the "Connect" button next to them?

**Comment by seal-82** [comment]
In case anybody comes across this thread (like me) looking for a solution to auto-configure calendar and address books in Thunderbird, there is one important thing to consider to make it work:

You have to set the additional DNS server entries like mentioned in the mailcow documentation: https://docs.mailcow.email/getstarted/prerequisite-dns/#the-advanced-dns-configuration

Especially the entries for caldavs and carddavs. This made it work for me... :-)

**Comment by apio-sys** [comment]
This is not an issue. Works out of the box with the DNS entries (normal, not additional) as per documentation.

**Comment by seal-82** [comment]
Tested it with a user of a domain not having the additional DNS server entries. For this mailbox, Thunderbird did not find the calendar and address books. Strange...
This is definitively not an issue, but I wanted to let people (like me) know where to look for a solution.
